addons/web/static/tests/views/fields/daterange_field_tests.js

```diff
@@ -466,6 +466,66 @@ QUnit.module("Fields", (hooks) => {
         }
     );
 
+    QUnit.test(
+        "Daterange field keyup should not erase end date [REQUIRE FOCUS]",
+        async function (assert) {
+            assert.expect(4);
+
+            serverData.models.partner.fields.datetime_end = {
+                string: "Datetime End",
+                type: "datetime",
+            };
+            serverData.models.partner.records[0].datetime_end = "2017-03-13 00:00:00";
+
+            await makeView({
+                type: "form",
+                resModel: "partner",
+                serverData,
+                arch: `
+                    <form>
+                        <field name="datetime" widget="daterange" options="{'related_end_date': 'datetime_end'}"/>
+                        <field name="datetime_end" widget="daterange" options="{'related_start_date': 'datetime'}"/>
+                    </form>`,
+                resId: 1,
+            });
+
+            // check date display correctly in readonly
+            assert.strictEqual(
+                target.querySelector(".o_field_daterange input").value,
+                "02/08/2017 15:30:00",
+                "the start date should be correctly displayed"
+            );
+            assert.strictEqual(
+                target.querySelectorAll(".o_field_daterange input")[1].value,
+                "03/13/2017 05:30:00",
+                "the end date should be correctly displayed"
+            );
+
+            // reveal the daterangepicker
+            await click(target.querySelector(".o_field_daterange[name='datetime'] input"));
+
+            // the keyup event should not be handled by daterangepicker
+            await triggerEvent(target, ".o_field_daterange[name='datetime'] input", "keyup", {
+                key: "ArrowLeft",
+            });
+
+            const datepickers = document.querySelectorAll(`.daterangepicker`);
+            await click(datepickers[1].querySelector(".applyBtn"));
```

Comment:
add a comment telling that we should check that the range is still correctly displayed in the picker (TODO), jum will add the assertions in his PR where he introduces our version of daterangepicker

Reply:
or directly to the assertions if the dom, he'll adapt them